src/main/kotlin/com/exactpro/th2/http/client/util/MessageUtil.kt

```diff
@@ -100,6 +100,18 @@ private fun createRequest(head: Message, body: RawMessage): RawHttpRequest {
     val parentEventId = head.parentEventId.id.ifEmpty { body.parentEventId.id }
     val metadataProperties = body.metadata.propertiesMap + head.metadata.propertiesMap
 
+    head.metadata.propertiesMap.forEach{it ->
+        val name = it.key
+        val value = it.value
+
+        if (!(name.isNullOrEmpty() || value.isNullOrEmpty())) {
+            if (name.startsWith("header-", ignoreCase = true)) {
```

Comment:
```suggestion
        if (value.isNotBlank() && name.startsWith("header-", ignoreCase = true)) {
```

Null-checks are redundant here because null key/value cannot be added into metadata property map

> "header-"

Create a constant for this and use its length for substring call below